### PR TITLE
Starkmatt/c02 network01

### PR DESCRIPTION
**Pull request by starkmatt**
# Description

Exercise: c02-network01 

Student Name: Matt G

A brief explanation of what has been done.
Had a book out and tried to do every single calculation on paper so I could understand the process.
I'll need to review the Hosts calculation I think

## Type of change

- [x] Exercise Submission
- [ ] Class content (instructors/admins)
- [ ] Documentation update
- [ ] Repo management

### CODE REVIEW

File: classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+IP Address in Binary: 11111001.10100101.10100110.10000111
+
+Subnet in Binary: 11111111.11111111.11111111.11111100
+
+Subnet in Decimals: 255.255.255.252
+
+Private / Public: Private
+
+IP Addresses for hosts available (AWS):
+2 (Unsure if my calculations were correct)
```
Comment (denstorti) [inline classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt]: that would be 4, as we have 2^2 = 4. But in AWS, the minimum subnet size is /28, so a /30 subnet is not valid in AWS.

File: classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+IP Address in Binary: 11101100.01000100.11100000.00010010
+
+Subnet in Binary 11111111.11111111.11111111.11111111
+
+Subnet in Decimals: 255.255.255.255
+
+Private / Public:Public
+
+IP Addresses for hosts available (AWS): 0 (or 1 including my connection ?)
```
Comment (denstorti) [inline classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt]: that would be 1, as we have 2^0 = 1. But in AWS, the minimum subnet size is /28, so a /30 subnet is not valid in AWS.

File: classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+IP Address in Binary: 11000000.10101000.00000000.00000001
+
+Subnet in Binary: 11111111.11111111.00000000.00000000
+
+Subnet in Decimals:255.255.0.0
+
+Private / Public: Private
+
+IP Addresses for hosts available (AWS): 65,534
```
Comment (caiocezart) [inline classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt]: are you sure on the 65,534?

File: classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt (hunk 1 above)
Comment (caiocezart) [reply classes/02class/exercises/c02-network01/starkmatt/c02-network01.txt]: actually just 2 valid hosts, 4 ip counts less 2 (network + broadcast) but as stated by Denis, AWS required minimum /28 as they need 5 ips

**Comment by caiocezart** [comment]
no need for this PR anymore, your code got merged through another one. Feel free to close it

**Comment by starkmatt** [comment]
<strong>Not needed anymore!</strong>

**Pull request closed without merging**